smbmap: add page (#4749)
tldr-pages/tldr: pages/linux/smbmap.md

# smbmap

> SMB enumeration tool.
> More information: <https://github.com/ShawnDEvans/smbmap>.

> - Enumerate SMB shares and permissions on a host:

Reviewer: I prefer the verb display, if this command print information in the terminal:
```suggestion
- Display SMB shares and permissions on a host:
```
The same for the others examples.

Author: OK, adapted.

> `smbmap -u {{username}} -p {{password}} -H {{ip}}`

Reviewer 2: Specifying a password like this will mean that the password ends up in the user's shell history. Is there another way we can pass the password to avoid this? Same goes for all the other examples.

Author: This tool is typically used during penetration tests. And indeed, you should carefully manage your history while using cleartext passwords in command lines. That's the way it is used in general.
But this is a very good remark ; looking at the source of the tool reveals that you can either use `-p {{password}}` (to specify the password in the command line) or `--prompt` (to prompt for a password using the `getpass.getpass()` function. I will adapt at least one example with the prompt option.

Author: I adapted one example. Good for you ?

Reviewer 2: Heya! Ah, I see. That looks much better! Can this be reflected across the other examples please?

Author: Done.

- Display SMB shares and permissions on a host, specifying the domain and passing the password NTLM hash:

`smbmap -u {{username}} -p {{ntlm_hash}} -d {{domain}} -H {{ip}}`

- Display SMB shares and list a single level of directories and files:

`smbmap -u {{username}} -p {{password}} -H {{ip}} -r`

- Display SMB shares and recursively list a defined number of levels of directories and files:

`smbmap -u {{username}} -p {{password}} -H {{ip}} -R --depth {{3}}`

- Display SMB shares and recursively list directories and files, downloading the files matching a regular expression:

`smbmap -u {{username}} -p {{password}} -H {{ip}} -R -A {{pattern}}`

- Display SMB shares and recursively list directories and files, searching for file content matching a regular expression:

`smbmap -u {{username}} -p {{password}} -H {{ip}} -R -F {{pattern}}`

> - Execute an OS command on a remote system:

Reviewer 2: Perhaps this will read better?
```suggestion
- Execute a shell command on a remote system:
```

`smbmap -u {{username}} -p {{password}} -H {{ip}} -x {{command}}`

> - Upload a file on a remote system:

Reviewer 2: Perhaps this will read better?
```suggestion
- Upload a file to a remote system:
```
